File: 0283-move-zeroes/solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int n = nums.size();
        
        for(int key,j,i=0;i<n;++i){
            j = i;
            key = nums[i];
            while((j > 0) && (nums[j-1] == 0)){
                nums[j] = nums[j-1];
                j--;
            }
            nums[j] = key;
        }
    }
};

File: 0283-move-zeroes/solution.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        List<int> arr = new List<int>();

        for(int i=0; i<nums.Length; i++){
            if(nums[i] != 0){
                arr.Add(nums[i]);
            } 
        }

        arr.AddRange(Enumerable.Repeat(0, nums.Length - arr.Count).ToList());
        
        for(int i = 0; i<nums.Length; i++){
            nums[i] = arr[i];
        }
    }
}
